presto-hive/src/test/java/com/facebook/presto/hive/s3/TestPrestoS3FileSystem.java

```diff
@@ -793,6 +797,44 @@ public ObjectListing listObjects(ListObjectsRequest listObjectsRequest)
         }
     }
 
+    @Test
+    public void testMultipart()
+            throws IOException, URISyntaxException
+    {
+        try (PrestoS3FileSystem fs = new PrestoS3FileSystem()) {
+            MockAmazonS3 s3 = new MockAmazonS3();
+            Configuration conf = new Configuration();
+            conf.setLong(S3_MULTIPART_MIN_PART_SIZE, 512L);
+            fs.initialize(new URI("s3n://test-bucket/"), conf);
+            fs.setS3Client(s3);
+            try (FSDataOutputStream out = fs.create(new Path("test"))) {
+                out.flush();
+            }
+            // since we didn't exceed the part size, no multipart requests should be sent
+            assertEquals(s3.initiateMultipartRequests, 0);
+            assertEquals(s3.uploadPartRequests, 0);
+            assertEquals(s3.completePartUploadRequests, 0);
+            assertEquals(s3.putObjectRequests, 1);
+
+            try (FSDataOutputStream out = fs.create(new Path("test"))) {
+                out.write(new byte[512]);
+            }
+            // same as above
+            assertEquals(s3.initiateMultipartRequests, 0);
+            assertEquals(s3.uploadPartRequests, 0);
+            assertEquals(s3.completePartUploadRequests, 0);
+            assertEquals(s3.putObjectRequests, 2);
+
+            try (FSDataOutputStream out = fs.create(new Path("test"))) {
+                out.write(new byte[600]);
+                out.write(new byte[600]);
+            }
+            assertEquals(s3.initiateMultipartRequests, 1);
+            assertEquals(s3.uploadPartRequests, 2);
+            assertEquals(s3.completePartUploadRequests, 1);
+        }
```

Comment:
Maybe add a test that simulates error and subsequent cleanup?

Reply:
Added